chromium_src/third_party/blink/renderer/modules/webaudio/audio_buffer.cc

```diff
@@ -13,40 +13,30 @@
-      if (brave::OptionalAudioFarblingCallback audio_farbling_callback =      \
-              brave::BraveSessionCache::From(*context)                        \
-                  .GetAudioFarblingCallback(settings)) {                      \
-        DOMFloat32Array* destination_array = array.Get();                     \
-        size_t len = destination_array->length();                             \
-        if (len > 0) {                                                        \
-          float* destination = destination_array->Data();                     \
-          for (unsigned i = 0; i < len; ++i) {                                \
-            destination[i] = audio_farbling_callback->Run(destination[i], i); \
-          }                                                                   \
-        }                                                                     \
-      }                                                                       \
-    }                                                                         \
-  }
-
-#define BRAVE_AUDIOBUFFER_COPYFROMCHANNEL                                 \
+#define BRAVE_AUDIOBUFFER_GETCHANNELDATA                                  \
+  NotShared<DOMFloat32Array> array = getChannelData(channel_index);       \
   if (ExecutionContext* context = ExecutionContext::From(script_state)) { \
     if (WebContentSettingsClient* settings =                              \
             brave::GetContentSettingsClientFor(context)) {                \
-      if (brave::OptionalAudioFarblingCallback audio_farbling_callback =  \
-              brave::BraveSessionCache::From(*context)                    \
-                  .GetAudioFarblingCallback(settings)) {                  \
-        for (unsigned i = 0; i < count; i++) {                            \
-          dst[i] = audio_farbling_callback->Run(dst[i], i);               \
-        }                                                                 \
+      DOMFloat32Array* destination_array = array.Get();                   \
+      size_t len = destination_array->length();                           \
+      if (len > 0) {                                                      \
+        float* destination = destination_array->Data();                   \
+        brave::BraveSessionCache::From(*context).FarbleAudioChannel(      \
+            settings, destination, len);                                  \
       }                                                                   \
     }                                                                     \
   }
 
+#define BRAVE_AUDIOBUFFER_COPYFROMCHANNEL                                      \
+  if (ExecutionContext* context = ExecutionContext::From(script_state)) {      \
+    if (WebContentSettingsClient* settings =                                   \
+            brave::GetContentSettingsClientFor(context)) {                     \
```

Comment:
pass `ExecutionContext::From(script_state)` directly

Reply:
done

Reply:
I was wrong here. Context is actually used later, I've restored it in these macroses.